config/v1/types_feature.go

```diff
@@ -53,8 +53,12 @@ var (
 	// your cluster may fail in an unrecoverable way.
 	CustomNoUpgrade FeatureSet = "CustomNoUpgrade"
 
+	// OKD turns on features for OKD. Turning this feature set ON is supported for OKD clusters, but NOT for OpenShift clusters
+	// this feature set on CANNOT BE UNDONE for OKD clusters and when enabled on OpenShift clusters it PREVENTS UPGRADES.
+	OKD FeatureSet = "OKD"
```

Comment:
How do we prevent this from being an option on OCP clusters?

What options do we have for including things in OCP but not OKD? Does OKD ship as a different cluster profile perhaps?

Reply:
yes we need to do this. similar to the model that the[ installer](https://github.com/openshift/installer/blob/main/pkg/types/installconfig_scos.go) and other repos follow, we can restrict it to be enabled only on OKD clusters.

Reply:
> can we do this in cluster config operator maybe?

We used to validate the feature gate transitions and values in the API server directly, but this was removed in favour of CEL, see https://github.com/openshift/kubernetes/pull/1944/files

Config operator doesn't have any admission time validation ability

> Why do we want to prevent it from being turned on for OCP?

I don't think we want an extra feature set in OCP. We have TPNU, we have DPNU, we have CNU.

Enabling an OKD feature sat that isn't OKDNU in OCP sends the wrong signal IMO. All of our features sets that have existed in OCP have either been upgradable, or have had `NoUpgrade` in the name.

The OKD set can be achieved as a version of CNU if a customer so desired.

Reply:
> We used to validate the feature gate transitions and values in the API server directly, but this was removed in favour of CEL, see https://github.com/openshift/kubernetes/pull/1944/files

Would bringing back validation like this: https://github.com/openshift/kubernetes/pull/2420 make sense?

Reply:
Do you want this featureset to be opt-in? Or on by default in all OKD clusters?

For HyperShift and OCP, they both have the `Default` featureset, but, these featuresets are not necessarily always identical. In fact, HyperShift for some time has been carrying additional gates that are not enabled in the standalone default set.

Personally, I think something like this would make more sense, but it is currently implemented using a cluster profile annotation that CVO uses to choose whether or not to install. I suspect if we were to work with the OTA team they'd have some options so that we could include one yaml file for OKD and one for standalone OCP

Reply:
@JoelSpeed what are your thoughts on the above? ^